### PR TITLE
Ensure original requirement is returned when parsing yarn.lock files

### PR DESCRIPTION
**Pull request by johnbintz-tidelift**
This will help us match up what Yarn versions came from what package.json requirements.

### CODE REVIEW

File: lib/bibliothecary/parsers/npm.rb [hunk 1]
```diff
@@ -70,6 +70,7 @@ def self.parse_yarn_lock(file_contents)
           {
             name: dep[:name],
             requirement: dep[:version],
+            original_requirement: dep[:requirement],
```
Comment (katzj) [inline lib/bibliothecary/parsers/npm.rb:73]: I'm getting a little hung up on the naming here and not loving this as it seems a little confusing. @jsonperl do you have any thoughts/suggestions?

**Review by katzj — COMMENTED** [review]
I suspect we're going to need this in more parsers (python definitely has some logic where it looks like the same idea would apply) but it's probably reasonable to add as we go 

Definitely feels like at least 6.9.0 worthy

File: lib/bibliothecary/parsers/npm.rb (hunk 1 above)
Comment (kbarrette) [reply lib/bibliothecary/parsers/npm.rb:73]: I have the feeling we already have a name for this in depci somewhere...

Comment (jsonperl) [reply lib/bibliothecary/parsers/npm.rb:73]: Hmm, so it's basically the requested version in the lockfile. lockfile_requirement?

**Review by katzj — APPROVED** [review]

**Review by jsonperl — APPROVED** [review]

**Pull request merged**